src/core_plugins/kibana/public/dashboard/directives/dashboard_panel.js

```diff
@@ -1,15 +1,19 @@
 import _ from 'lodash';
 import 'ui/visualize';
 import 'ui/doc_table';
-import { loadPanelProvider } from 'plugins/kibana/dashboard/components/panel/lib/load_panel';
+import 'plugins/kibana/dashboard/services/get_object_loaders_for_dashboard.factory';
 import FilterManagerProvider from 'ui/filter_manager';
 import uiModules from 'ui/modules';
 import panelTemplate from 'plugins/kibana/dashboard/components/panel/panel.html';
+import { getPersistedStateId } from 'plugins/kibana/dashboard/components/panel/lib/panel_state';
+import { loadSavedObject } from 'plugins/kibana/dashboard/components/panel/lib/load_saved_object';
 
 uiModules
 .get('app/dashboard')
-.directive('dashboardPanel', function (savedVisualizations, savedSearches, Notifier, Private, $injector) {
-  const loadPanel = Private(loadPanelProvider);
+.directive('dashboardPanel', function ($injector) {
+  const Private = $injector.get('Private');
```

Comment:
Yea, I think the idea is to stop using nameless functions that require Private, and explicitly register them in angular.  The more I think about this specific change though, the more I am unsure.

I was originally under the assumption that if I got rid of the parameters and only passed $Injector, I could avoid wrapping `dashboardPanel` with `Private` but that doesn't make sense because dashboardPanel is a directive, first off, and second off, is registered with angular, and no class registered with angular would require Private.

After going back to the original discussion with @w33ble I think the idea is to use $injector only for nameless functions, not for things already registered with angular.  Does that sound right, @w33ble?  So this should go back to it's previous version?